src/pools/abstract-pool.ts

```diff
@@ -135,6 +135,11 @@ export abstract class AbstractPool<
    */
   protected nextMessageId: number = 0
 
+  protected workerChoiceCallback: (
+    poolReference: IPool<Worker, Data, Response>,
+    ...args: unknown[]
```

Comment:
We could try to type this with `...T`
its new in TS since 4.0: https://devblogs.microsoft.com/typescript/announcing-typescript-4-0/#variadic-tuple-types